### PR TITLE
add flag to have CLI ignore files that cause errors

### PR DESCRIPTION
**Pull request by shabtaisharon**
for example:
./ds3_java_cli --http -c put_bulk -b testing -d dir --ignore-errors

### CODE REVIEW

**Comment by shabtaisharon** [comment]
I still need to add the tests

**Comment by shabtaisharon** [comment]
added test for --ignore-errors flag using --output-format cli & json

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Arguments.java [hunk 1]
```diff
@@ -578,4 +587,12 @@ void setNumberOfThreads(final String numberOfThreads) {
     public String getNumberOfThreads() {
         return this.numberOfThreads;
     }
+
+    public boolean isIgnoreErrors() {
+        return ignoreErrors;
+    }
+
+    public void setIgnoreErrors(final boolean ignoreErrors) {
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/Arguments.java]: This should be package private.  So, remove the `public` modifier.

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/PutBulk.java [hunk 2]
```diff
@@ -219,4 +219,64 @@ public SeekableByteChannel buildChannel(final String s) throws IOException {
...
+                            Utils.getFileName(inputDirectory, path),
+                            Utils.getFileSize(path)));
+                } catch (final IOException ex) {
+                    if (!ignoreErrors) throw ex;
+                    LOG.warn(String.format("WARN: file '%s' has an error and will be ignored", path.getFileName()));
+                    ignoredBuilder.add(new IgnoreFile(path, ex.toString()));
+                }
+            }
+
+            ignoredFiles = ignoredBuilder.build();
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/command/PutBulk.java]: Hmmm.  I typically don't encourage methods that have side effects like this.  It seems like this method should be static, and that it returns an object that has both a list of objects to act on, and the list of ignored files.

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/Utils.java [hunk 3]
```diff
@@ -38,7 +38,7 @@ public FileVisitResult visitFile(final Path file, final BasicFileAttributes attr
         return objectsBuilder.build();
     }
 
-    public static String getFileName(final Path rootDir, final Path filePath) {
+    public static String getFileName(final Path rootDir, final Path filePath) throws IOException{
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/Utils.java]: Was the `throws IOException` required?  If this worked before, I'd rather we not include it.

Comment (shabtaisharon) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/util/Utils.java]: it is required for the test since i'm throwing an exception for a corrupt file

**Comment by shabtaisharon** [comment]
I've removed the exception from getFileName and changed 2 test to throw exception when getFileSize is called

**Comment by shabtaisharon** [comment]
removed method side effects

**Comment by rpmoore** [comment]
This pull request has conflicts now.  Please update.

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/PutBulkView.java [hunk 4]
```diff
@@ -15,12 +15,46 @@
...
+import com.spectralogic.ds3cli.command.PutBulk;
 import com.spectralogic.ds3cli.models.PutBulkResult;
 
+import static com.spectralogic.ds3cli.util.Utils.nullGuard;
+
 public class PutBulkView implements View<PutBulkResult> {
     @Override
     public String render(final PutBulkResult result) {
-        return result.getResult();
+        if (result.getIgnoredFiles() == null || result.getIgnoredFiles().isEmpty()) {
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/cli/PutBulkView.java]: Shouldn't this use one of the Guard methods?

File: ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/json/PutBulkView.java [hunk 5]
```diff
@@ -24,6 +24,14 @@
     @Override
     public String render(final PutBulkResult result) throws JsonProcessingException {
         final CommonJsonView view = CommonJsonView.newView(CommonJsonView.Status.OK);
-        return JsonMapper.toJson(view.message(result.getResult()));
+
+        if (result.getIgnoredFiles() == null || result.getIgnoredFiles().isEmpty()) {
```
Comment (rpmoore) [inline ds3_java_cli/src/main/java/com/spectralogic/ds3cli/views/json/PutBulkView.java]: Shouldn't this use one of the Guard methods?

**Comment by shabtaisharon** [comment]
* Use Guard method
* Merging and fixing conflicts

**Pull request merged**